libs/components/modals/src/lib/modules/modal/modal-host.component.ts

```diff
@@ -109,15 +117,89 @@ export class SkyModalHostComponent implements OnDestroy {
       parent: this.injector,
     });
 
+    const modalsBefore = Array.from(
+      this.hostElement.nativeElement.parentElement.children
+    );
+
     const modalComponentRef = this.target.createComponent(
       factory,
       undefined,
       injector
     );
 
+    //const modalsAfter = this.hostElement.nativeElement.children;
+    const modalsAfter = Array.from(
+      this.hostElement.nativeElement.parentElement.children
+    );
+
+    // getting the element ref to the modal that was just opened
+    const modalElement = modalsAfter.filter(
+      (el) => modalsBefore.indexOf(el) < 0
+    )[0];
+
     modalInstance.componentInstance = modalComponentRef.instance;
 
+    // hidding all elements at the modal-host level from screenreaders
+    const ariaPreviousValueSiblings = new Map<Element, string | null>();
+    const siblings =
+      document.querySelector('sky-modal-host').parentElement.children;
+
+    for (let i = 0; i < siblings.length; i++) {
+      const sibling = siblings[i];
+
+      if (
+        sibling !== document.querySelector('sky-modal-host') &&
+        !sibling.hasAttribute('aria-live') &&
+        sibling.nodeName !== 'SCRIPT' &&
+        sibling.nodeName !== 'STYLE'
+      ) {
+        ariaPreviousValueSiblings.set(
+          sibling,
+          sibling.getAttribute('aria-hidden')
+        );
+
+        sibling.setAttribute('aria-hidden', 'true');
+      }
+    }
+
+    // hiding all sibling modals from screenreaders
+    const ariaPreviousValueChildren = new Map<Element, string | null>();
+    const children = document.querySelector('sky-modal-host').children;
+
+    for (let i = 0; i < children.length; i++) {
+      const child = children[i];
+
+      if (
+        child !== modalElement &&
+        !child.hasAttribute('aria-live') &&
+        child.nodeName !== 'SCRIPT' &&
+        child.nodeName !== 'STYLE'
+      ) {
+        ariaPreviousValueChildren.set(child, child.getAttribute('aria-hidden'));
+        child.setAttribute('aria-hidden', 'true');
+      }
+    }
+
     function closeModal() {
+      // unhiding modal-host siblings
+      ariaPreviousValueSiblings.forEach((previousValue, element) => {
```

Comment:
What happens if a lower modal closes before the most upper modal (not a common case - but could happen via the typescript service technically.

Reply:
im going to add a unit test that opens two modals and then closes the first one just to confirm that that is what would happen. but if that is so, im thinking make the list of aria hidden modals into a global variable and check that when any modal is closed.